StreamChatClient_v3/Models/ChannelModel.swift

```diff
@@ -132,7 +132,11 @@ extension ChannelModel {
 /// A convenience `ChannelModel` typealias with no additional channel data.
 public typealias Channel = ChannelModel<DefaultDataTypes>
 
-public protocol ChannelExtraData: Codable & Hashable {}
+/// Additional data fields `ChannelModel` can be extended with. You can use it to store your custom data related to a channel.
```

Comment:
How about: 
```swift
public protocol DefaultValueProvider {
    static var defaultValue: Self { get }
}

public protocol ChannelExtraData: Codable, Hashable, DefaultValueProvider {}
public protocol UserExtraData: Codable, Hashable, DefaultValueProvider {}
...
```

Reply:
Makes sense @evsaev 👍 I tweaked the naming a little bit and forced pushed since this touches a lot of commits.